tools/ensure_codegen_up_to_date.py

```diff
@@ -0,0 +1,64 @@
+"""
+ensure_codegen_up_to_date
+-------------------------
+Ensure changes to code generation are committed to repository
+"""
+
+import os
+import sys
+import subprocess
+
+def configure_git_user_email():
+    """
+    Configures git user email with dummy email id
+    """
+    if os.system('git config user.email "dummy@ni.com"') != 0:
+        print("Unable to configure \"user email\" using git", file=sys.stderr)
+        sys.exit(-1)
+
+def configure_git_user_name():
+    """
+    Configures git user email with dummy name
+    """
+    if os.system('git config user.name "dummy"') != 0:
+        print("Unable to configure \"user name\" using git", file=sys.stderr)
+        sys.exit(-1)
+
+def clean_codegen_files():
+    """
+    Before code generation clean the existing codegen files
+    """
+    if os.system("tox -e clean") != 0:
+        print("Unable to clean the repo using \"tox -e clean\"", file=sys.stderr)
+        sys.exit(-1)
+
+def create_codegen_files():
```

Comment:
prepend all helper functions with `_` which is a convention for "don't call this from the outside it's private".

Reply:
Updated